### PR TITLE
 test: Write integration tests for some releases commands using trycmd

### PR DESCRIPTION
**Pull request by kamilogorek**
This PR will act as the base for future integration testing.

### CODE REVIEW

**Review by AbhiPrasad — COMMENTED** [review]
Looks good, I like the snapshots! The fact that they are under `tests/cmd` instead of `tests/commands` is kind of confusing though. Maybe we put them under `tests/fixtures/commands`?

I'll comment for now, but feel free to ping anytime will slam the approval.

File: tests/common/mod.rs [hunk 1]
```diff
@@ -1,11 +1,59 @@
...
-    let mut env = HashMap::new();
-    env.insert(String::from("SENTRY_URL"), server_url());
-    env.insert(String::from("SENTRY_AUTH_TOKEN"), String::from("lolnope"));
-    env.insert(String::from("SENTRY_ORG"), String::from("wat-org"));
-    env.insert(String::from("SENTRY_PROJECT"), String::from("wat-project"));
-    env
+use mockito::{mock, server_url, Matcher, Mock};
+use trycmd::TestCases;
+
+pub const UTC_DATE_FORMAT: &str = r#"\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{6}Z"#;
```
[the remark this answers is not in the document]
Comment (kamilogorek) [reply tests/common/mod.rs:4]: I was thinking about this, but I want to reuse it across different regexes. We don't know what key will hold the timestamp in front.

File: tests/common/mod.rs [hunk 2]
```diff
@@ -1,11 +1,59 @@
...
+
+pub fn create_testcase() -> TestCases {
+    let test_case = TestCases::new();
+    test_case
+        .env("SENTRY_URL", server_url())
```
[the remark this answers is not in the document]
Comment (kamilogorek) [reply tests/common/mod.rs:9]: `.env` is returning `&TestCases` and we need to return owned data here.

File: tests/cmd/releases/releases-finalize-hyphen.trycmd [hunk 3]
```diff
@@ -0,0 +1,6 @@
+```
+$ sentry-cli releases finalize -wat-release
+? success
+Finalized release wat-release.
```
[the remark this answers is not in the document]
Comment (kamilogorek) [reply tests/cmd/releases/releases-finalize-hyphen.trycmd:4]: Its unfortunately based on the server response for now, and I didn't want to create 2 API mocks :3

File: tests/cmd/releases/releases-finalize-hyphen.trycmd [hunk 4]
```diff
@@ -0,0 +1,6 @@
+```
+$ sentry-cli releases finalize -wat-release
```
[the remark this answers is not in the document]
Comment (kamilogorek) [reply tests/cmd/releases/releases-finalize-hyphen.trycmd:2]: Ok

**Comment by kamilogorek** [comment]
Updated:

- changed structure to:
```
tests/integration
 - <command>
 - _api/<command>
 - _cases/<command>
```
- removed `common` in favor of putting all helpers inside `integration/mod.rs`
- renamed `create_testcase` to `register_test` and make it format path to the test itself (no more `tests/integration/_cases/` needed)
- made `with_response_file` format path to the response itself (no more `tests/integration/_api/` needed)

**Pull request merged**